tests/rptest/tests/offset_retention_test.py

```diff
@@ -181,3 +183,77 @@ def offsets_exist():
         # after waiting for twice the retention period, it should be gone
         time.sleep(self.period * 2 - self.period / 2)
         assert not offsets_exist()
+
+
+class OffsetDeletionTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3), )
+    group = "hey_hey_group"
+
+    def __init__(self, test_context):
+        super(OffsetDeletionTest, self).__init__(test_context=test_context)
+
+        self.rpk = RpkTool(self.redpanda)
+        self.kcl = KCL(self.redpanda)
+
+    @cluster(num_nodes=4)
+    def test_offset_deletion(self):
+        def offsets_exist():
+            desc = self.rpk.group_describe(self.group)
+            return len(desc.partitions) > 0
+
+        # Produce some data to a topic and consume assigning new consumer group
+        for _ in range(0, 10):
+            self.rpk.produce(self.topic, "k", "v", partition=0)
+            self.rpk.produce(self.topic, "k", "v", partition=1)
+            self.rpk.produce(self.topic, "k", "v", partition=2)
+        self.rpk.consume(self.topic, n=3, group=self.group)
+        wait_until(offsets_exist, timeout_sec=1, backoff_sec=1)
+
+        # Assert offset-delete errors when request contains missing topic-partitions
+        missing_topic = f"{self.topic}-foo"
+        missing_topic_partitions = {missing_topic: [0, 1, 2]}
```

Comment:
Test another error case of mixed cases, i.e. some topic/partitions are valid, some cause errors to be returned